packages/rum-core/src/domain/startCustomerDataTelemetry.ts

```diff
@@ -0,0 +1,145 @@
+import type { BatchFlushEvent, Context, ContextManager, Observable, Telemetry } from '@datadog/browser-core'
+import {
+  isEmptyObject,
+  includes,
+  isExperimentalFeatureEnabled,
+  performDraw,
+  ONE_SECOND,
+  addTelemetryDebug,
+  monitor,
+} from '@datadog/browser-core'
+import { RumEventType } from '../rawRumEvent.types'
+import type { RumEvent } from '../rumEvent.types'
+import type { RumConfiguration } from './configuration'
+import type { FeatureFlagContexts } from './contexts/featureFlagContext'
+import type { LifeCycle } from './lifeCycle'
+import { LifeCycleEventType } from './lifeCycle'
+
+export const MEASURES_PERIOD_DURATION = 10 * ONE_SECOND
+
+type Measure = {
+  min: number
+  max: number
+  sum: number
+}
+
+type CurrentPeriodMeasures = {
+  batchCount: number
+  batchBytesCount: Measure
+  batchMessagesCount: Measure
+  globalContextBytes: Measure
+  userContextBytes: Measure
+  featureFlagBytes: Measure
+}
+
+type CurrentBatchMeasures = {
+  globalContextBytes: Measure
+  userContextBytes: Measure
+  featureFlagBytes: Measure
+}
+
+let currentPeriodMeasures: CurrentPeriodMeasures
+let currentBatchMeasures: CurrentBatchMeasures
+
+export function startCustomerDataTelemetry(
+  configuration: RumConfiguration,
+  telemetry: Telemetry,
+  lifeCycle: LifeCycle,
+  globalContextManager: ContextManager,
+  userContextManager: ContextManager,
+  featureFlagContexts: FeatureFlagContexts,
+  batchFlushObservable: Observable<BatchFlushEvent>
+) {
+  const customerDataTelemetrySampleRate =
+    telemetry.enabled && performDraw(configuration.customerDataTelemetrySampleRate)
+  if (!customerDataTelemetrySampleRate || !isExperimentalFeatureEnabled('customer_data_telemetry')) {
+    return
+  }
+
+  initCurrentPeriodMeasures()
+  initCurrentBatchMeasures()
+
+  // We measure the data of every view updates even if there could only be one per batch due to the upsert
+  // It means that contexts bytes count sums can be higher than it really is
+  lifeCycle.subscribe(LifeCycleEventType.RUM_EVENT_COLLECTED, (event: RumEvent & Context) => {
+    if (!isEmptyObject(globalContextManager.get())) {
+      updateMeasure(currentBatchMeasures.globalContextBytes, globalContextManager.getBytesCount())
+    }
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏In case of a `clearContext` should not we update the measure to take into account the empty context?